rules/job.go

```diff
@@ -58,16 +75,23 @@ outer:
 				continue
 			}
 
-			if matches(rule, component) {
+			if matches(rule, *component) {
 				logger.Infof("Rule %s matched component %s", rule, component)
 
 				incident := rule.Template
 				incident.IncidentRuleID = _rule.ID
-				incident.Status = "open"
+				incident.Status = api.IncidentStatusOpen
+				incident.Severity = "Low" // TODO: make this configurable
 				if incident.Type == "" {
 					incident.Type = api.IncidentTypeAvailability
 				}
 				incident.Title = component.Name + " is " + string(component.Status)
+
+				if _, ok := openIncidentsMap[_rule.ID.String()][incident.Title]; ok {
```

Comment:
I am checking duplicates by the incident title. Couldn't find a better alternative.

I think we should probably add component id to the incident as well? 
- That'll be much more robust in determining duplicates specially if the incident title template changes.
- Also, if a rule creates an incident when the component status is "error" and then later when the status changes to "warn", we'll have 2 different incidents. If we have component id on the incident, then we can close the old one before creating a new incident.

Reply:
Can we not do a lookup on the evidence table ?

Reply:
There could be a lot of components and a lookup for each one of them can potentially be a lot, so I just did a bulk query once.